lib/src/list/list_builder.dart

```diff
@@ -38,9 +38,11 @@ class ListBuilder<E> {
   /// The `ListBuilder` can be modified again and used to create any number
   /// of `BuiltList`s.
   BuiltList<E> build() {
-    if (_builtList != null) return _builtList;
-    _copyBeforeWrite = true;
```

Comment:
Below whenever you set _copyBeforeWrite, you also always need to set _builtList to null.